jaeger-core/src/main/java/com/uber/jaeger/senders/HttpSender.java

```diff
@@ -24,61 +24,81 @@
 
 import com.uber.jaeger.thriftjava.Batch;
 import com.uber.jaeger.thriftjava.Process;
+import com.uber.jaeger.thriftjava.Span;
 import java.io.IOException;
-import java.net.URI;
-import java.net.URISyntaxException;
 import java.util.List;
-import org.apache.http.HttpResponse;
-import org.apache.http.HttpStatus;
-import org.apache.http.client.HttpClient;
-import org.apache.http.client.methods.HttpPost;
-import org.apache.http.entity.ByteArrayEntity;
-import org.apache.http.impl.client.DefaultHttpClient;
+import okhttp3.HttpUrl;
+import okhttp3.MediaType;
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import okhttp3.RequestBody;
+import okhttp3.Response;
 import org.apache.thrift.TException;
 import org.apache.thrift.TSerializer;
-import org.apache.thrift.protocol.TBinaryProtocol.Factory;
+import org.apache.thrift.protocol.TBinaryProtocol;
 import org.apache.thrift.protocol.TProtocolFactory;
 
 public class HttpSender extends ThriftSender {
+
   private static final String HTTP_COLLECTOR_JAEGER_THRIFT_FORMAT_PARAM = "format=jaeger.thrift";
-  private static final TProtocolFactory PROTOCOL_FACTORY = new Factory();
+  private static final TProtocolFactory PROTOCOL_FACTORY = new TBinaryProtocol.Factory();
   private static final TSerializer SERIALIZER = new TSerializer(PROTOCOL_FACTORY);
+  private static final int ONE_MB_IN_BYTES = 1048576;
+  private static final MediaType MEDIA_TYPE_THRIFT = MediaType.parse("application/x-thrift");
+  private final OkHttpClient httpClient;
+  private final HttpUrl collectorUrl;
 
-  private final HttpClient httpClient = new DefaultHttpClient();
-  private final URI collectorUri;
+  /**
+   * @param endpoint Jaeger REST endpoint consuming jaeger.thrift, e.g
+   * http://localhost:14268/api/traces
+   *
+   * Uses the default {@link okhttp3.OkHttpClient} which uses {@link okhttp3.ConnectionPool#ConnectionPool()}.
+   * Use {@link HttpSender#HttpSender(java.lang.String, int, okhttp3.OkHttpClient)} to adjust parameters.
+   */
+  public HttpSender(String endpoint) {
+    this(endpoint, ONE_MB_IN_BYTES, new OkHttpClient());
+  }
 
   /**
-   * @param endpoint Jaeger REST endpoint consuming jaeger.thrift, e.g http://localhost:14268/api/traces
-   * @param maxPacketSize max packet size
+   * @param endpoint Jaeger REST endpoint consuming jaeger.thrift, e.g
+   * http://localhost:14268/api/traces
+   * @param maxPayloadBytes max bytes to serialize as payload
+   * @param client a client used to make http requests
    */
-  public HttpSender(String endpoint, int maxPacketSize) {
-    super(PROTOCOL_FACTORY, maxPacketSize);
-    this.collectorUri = constructCollectorUri(endpoint);
+  public HttpSender(String endpoint, int maxPayloadBytes, OkHttpClient client) {
```

Comment:
> You are effectively breaking the compatibility with older clients. Is that intentional? 

@jpkrohling  It is because I assumed nobody was using this (especially with #216). I could do what you suggest, and mark it deprecated.

>+1 - passing okhttpclient to a public constructor is a functionality nobody asked for, thus is unnecessary increase in the API surface. 

@yurishkuro This is true, I only exposed it because it allows end users to easily change how data is sent to collectors. It also allows for easy configuration of authentication, proxies, etc. 
To minimize the API surface, we could simply ask for easy of these configuration options when somebody requests for it, and then pass in parameters to okhttp builder.